fix: Talos health checks for Kubernetes v1.32+

mlinares1998 opened the pull request:
Hi! 👋

Talos Kubernetes registry support is deprecated starting in Kubernetes v1.32 due to the AuthorizeNodeWithSelectors feature gate being enabled by default.

To maintain cluster health checks in preparation to the module v2.x.x with Talos v1.9/K8s v1.32 , this PR enables the public SideroLabs discovery service when the target K8s version is equal or higher than v1.32. 
According to the documentation, this approach remains secure for cluster owners, as all data is encrypted using AES-GCM and can only be decrypted by the nodes themselves.

I have managed to successfully update to Talos v1.9/K8s v1.32 with this change with healthchecks working correctly.

Reference:
https://github.com/siderolabs/talos/issues/9980
https://www.talos.dev/v1.9/talos-guides/discovery/#kubernetes-registry 
https://www.talos.dev/v1.9/talos-guides/discovery/#discovery-service-registry

M4t7e commented:
Thanks a lot, @mlinares1998 🙂 This problem has also been discovered in #51

I would like to avoid version checks if possible and I assume some users may not want Talos to continuously "call home". With Talos 1.9 and Kubernetes 1.32 coming in the next major release, I'd suggest making service discovery configurable instead. We could enable Talos service discovery by default and disable Kubernetes-based discovery.

Here’s what I propose:

```hcl
variable "talos_discovery_kubernetes_enabled" {
  description = "Enable or disable Kubernetes-based discovery for Talos. Deprecated as of Kubernetes v1.32, where the AuthorizeNodeWithSelectors feature gate is enabled by default."
  type        = bool
  default     = false
}

variable "talos_discovery_service_enabled" {
  description = "Enable or disable Talos discovery service."
  type        = bool
  default     = true
}
```

I’d like to introduce this ahead of the upgrade so that service discovery is already in place and health checks don’t fail during the upgrade.

mlinares1998 commented:
Hi @M4t7e 
I agree with your idea. :smile: 
I have removed the version check and added the variables as you said in https://github.com/hcloud-k8s/terraform-hcloud-kubernetes/pull/101/commits/e7fab272e6b0a39539c12aa1910227a7b339780a
I have also added further information in the readme about this change.

Regards!.
